docs/developer/architecture/ci.rst

```diff
@@ -0,0 +1,155 @@
+Continuous Testing
+==================
+
+
+This document will not describe how to write a test but just the list of tests
+that should be done and when.
+
+The goal is to:
+
+- have day-to-day development and PRs merged faster
+- have a great test coverage
+
+Lets define 2 differents stages of continuous testing:
+
+- Pre-merge: Run during development process on changes not yet merged
+- Post-merge: Run on changes already approved and merged in development
+  branches
+
+Pre-merge
+---------
+
+What should be tested in pre-merge on every branch used during development
+(``user/*``, ``feature/*``, ``improvement/*``, ``bugfix/*``, ``w/*``).
+The pre-merge test should not long too much time (less than 40 minutes
+is great) so we can't test everything in pre-merge, we should only test
+building of the product and check that product still usable.
+
+- Building tests
+
+  - Build
+  - Lint
+  - Unit tests
+
+- Installation tests
+
+  - Simple install RHEL
+  - Simple install CentOs + expansion
+
+When merging several pull requests at the same time, given that we are on a
+queue branch (``q/*``), we may want to have or we may require additional tests
+as combinaison of several PRs could have larger impact than all individual PR:
```

Comment:
`When merging several pull requests at the same time, given that we are on a
queue branch (``q/*``), we may require additional tests
as a combination of several PRs could have a larger impact than all individual PR:`